pulpcore/cli/common/generic.py

```diff
@@ -133,7 +133,8 @@ def __init__(
     ) -> None:
         self.username = api_kwargs.pop("username", None)
         self.password = api_kwargs.pop("password", None)
-        api_kwargs["auth_provider"] = PulpCLIAuthProvider(pulp_ctx=self)
+        if self.username:
```

Comment:
Can we use `if not self.cert` here instead?
The PulpCLIAuthProvider is supposed to prompt for the username if we don't have one, if I remember correctly.

Reply:
```
AttributeError: 'PulpCLIContext' object has no attribute 'cert'
```

:woman_shrugging:

Reply:
```
if not api_kwargs.get('cert', None):
```
that seems to work?

Reply:
wait, no, that's still broken as hell. `pulp status` works, but other things (that actually need auth):
```
# pulp user list
Traceback (most recent call last):
  File "/bin/pulp", line 8, in <module>
    sys.exit(main())
             ^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
         ^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulpcore/cli/common/generic.py", line 290, in invoke
    return super().invoke(ctx)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulpcore/cli/common/generic.py", line 290, in invoke
    return super().invoke(ctx)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
                           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulpcore/cli/common/generic.py", line 290, in invoke
    return super().invoke(ctx)
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/decorators.py", line 84, in new_func
    return ctx.invoke(f, obj, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulpcore/cli/common/generic.py", line 1252, in callback
    result = entity_ctx.list(limit=limit, offset=offset, parameters=kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulp_glue/common/context.py", line 767, in list
    result: t.Mapping[str, t.Any] = self.call("list", parameters=payload)
                                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulp_glue/common/context.py", line 704, in call
    return self.pulp_ctx.call(
           ^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulp_glue/common/context.py", line 357, in call
    result = self.api.call(
             ^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulp_glue/common/openapi.py", line 687, in call
    request: requests.PreparedRequest = self.render_request(
                                        ^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/site-packages/pulp_glue/common/openapi.py", line 592, in render_request
    if security and self.auth_provider:
                    ^^^^^^^^^^^^^^^^^^
AttributeError: 'OpenAPI' object has no attribute 'auth_provider'
```

Reply:
Okay, explicitly setting it to `None` in that case works.

- cert auth works fine.
- not providing any auth yields `Error: No suitable auth scheme found.` when doing `pulp user list`
- password auth seems not to work, but I *think* that's Katello's setup doing

Reply:
Hm. It only offers me `cookieAuth`, not `basic` :woman_shrugging: 
```
{'cookieAuth': {'type': 'apiKey', 'in': 'cookie', 'name': 'sessionid'}}
```

Reply:
Actually the PulpRemoteUserAuthentication should (allow to) add "mutualTLS" to the security schemata.
But yes, that means no BasicAuth is allowed (at least not terminated by Django).